src/distributed_ls/Euclid/ilu_mpi_bj.c

```diff
@@ -173,16 +173,16 @@ HYPRE_Int symbolic_row_private(HYPRE_Int localRow, HYPRE_Int beg_row, HYPRE_Int
 {
   START_FUNC_DH
   HYPRE_Int level = ctx->level, m = ctx->F->m;
-  HYPRE_Int *cval = ctx->F->cval, *diag = ctx->F->diag, *rp = ctx->F->rp; 
+  HYPRE_Int *cval = ctx->F->cval, *diag = ctx->F->diag, *rp = ctx->F->rp;
   HYPRE_Int *fill = ctx->F->fill;
   HYPRE_Int count = 0;
   HYPRE_Int j, node, tmp, col, head;
   HYPRE_Int fill1, fill2;
-  float val;
```

Comment:
Do you know why did they specifically use single precision?

Reply:
I don't know, however this variable is used in https://github.com/hypre-space/hypre/blob/efe270582b8111877a0fc58ef849f04f62988afb/src/distributed_ls/Euclid/ilu_mpi_bj.c#L196 and `AVAL` is a `HYPRE_Real` (see function's definition). So, I figured out `val` should be a `HYPRE_Real`.